packages/caravan-fees/src/types.ts

```diff
@@ -0,0 +1,444 @@
+   */
+  network: Network;
+
+  /**
+   * The dust threshold in satoshis. Outputs below this value are considered "dust"
+   * and may not be economically viable to spend.
+   */
+  dustThreshold: Satoshis;
+
+  /**
+   * The type of script used for the transaction (e.g., P2PKH, P2SH, P2WSH).
+   */
+  scriptType: string;
+
+  /**
+   * The number of required signers for the multisig setup.
+   */
+  requiredSigners: number;
+
+  /**
+   * The total number of signers in the multisig setup.
+   */
+  totalSigners: number;
+
+  /**
+   * The target fee rate in satoshis per virtual byte. This is used to calculate
+   * the appropriate fee for the transaction.
+   */
+  targetFeeRate: FeeRateSatsPerVByte;
+
+  /**
+   * The absolute fee of the original transaction in satoshis.
+   */
+  absoluteFee: Satoshis;
+
+  /**
+   * Whether to attempt full RBF even if the transaction doesn't signal it.
+   * @default false
+   */
+  fullRBF?: boolean;
+
+  /**
+   * If true, enforces stricter validation rules.
+   * @default false
+   */
+  strict?: boolean;
+}
+
+/**
+ * Options for creating an accelerated RBF transaction.
+ */
+export interface AcceleratedRbfOptions
+  extends Omit<CancelRbfOptions, "cancelAddress"> {
+  /**
+   * The index of the change output in the original transaction.
+   */
+  changeIndex?: number;
+
+  /**
+   * The address to use for the new change output, if different from the original.
+   */
+  changeAddress?: string;
+}
+
+/**
+ * Options for creating a CPFP transaction.
+ */
+export interface CPFPOptions {
+  /**
+   * The hex-encoded original (parent) transaction to be accelerated.
+   */
+  originalTx: string;
+
+  /**
+   * Array of available UTXOs, including the spendable output from the parent transaction.
+   */
+  availableInputs: UTXO[];
+
+  /**
+   * The index of the output in the parent transaction that will be spent in the child transaction.
+   */
+  spendableOutputIndex: number;
+
+  /**
+   * The address where any excess funds (change) will be sent in the child transaction.
+   */
+  changeAddress: string;
+
+  /**
+   * The Bitcoin network being used (e.g., mainnet, testnet).
+   */
+  network: Network;
+
+  /**
+   * The dust threshold in satoshis. Outputs below this value are considered "dust".
+   */
+  dustThreshold: Satoshis;
+
+  /**
+   * The type of script used for the transaction (e.g., P2PKH, P2SH, P2WSH).
+   */
+  scriptType: string;
+
+  /**
+   * The target fee rate in satoshis per virtual byte. This is used to calculate
+   * the appropriate fee for the transaction.
+   */
+  targetFeeRate: FeeRateSatsPerVByte;
+
+  /**
+   * The absolute fee of the original transaction in satoshis.
+   */
+  absoluteFee: Satoshis;
+
+  /**
+   * The number of required signers for the multisig setup.
+   */
+  requiredSigners: number;
+
+  /**
+   * The total number of signers in the multisig setup.
+   */
+  totalSigners: number;
+
+  /**
+   * If true, enforces stricter validation rules.
```

Comment:
can you give some examples of validation rules

Reply:
changes [done](https://github.com/caravan-bitcoin/caravan/pull/114/commits/ed9f6c70ed681c17d1ed07b444329b98811948db#diff-593f8d602f801c56261ed7a8dc941947102f6a7b26c483c5ee8cf9c293fffb33R441)